pkgs/build-support/trivial-builders/default.nix

```
@@ -362,6 +362,59 @@ rec {
         else checkPhase;
     };
 
+  /*
+    Similar to writeShellApplication but runs the provided text through argbash
+    to generate an options parser before using writeShellApplication under the hood.
+
+    Example:
+
+    Writes my-file to /nix/store/<store path>/bin/my-file and makes executable.
+
+    > my-file --help
+    A cow greets you from an argbash-generated, nix-packaged script.
+    Usage: my-file [-n|--name <arg>] [-h|--help]
+      -n, --name: The name to greet (default: 'Lev')
+      -h, --help: Prints help
+
+    writeShellApplication {
+      name = "my-file";
+      runtimeInputs = [ cowsay ];
+      text = ''
+        # ARG_OPTIONAL_SINGLE([name], [n], [The name to greet], [Lev])
+        # ARG_HELP([A cow greets you from an argbash-generated, nix-packaged script.])
+        # ARGBASH_GO
+        # [ <-- needed because of Argbash
+
+        cowsay "Hello, $_arg_name!"
+
+        # ] <-- needed because of Argbash
+       '';
+    }
+
+  */
+  writeArgbashShellApplication =
+    { name
+    , text
+    , runtimeInputs ? [ ]
+    , checkPhase ? null
+    }:
```

Let `writeShellApplication` verify the arguments.

```suggestion
    , ...
    }@args:
```